indica/readers/st40reader.py

```diff
@@ -269,26 +280,34 @@ def _get_helike_spectroscopy(
             "machine_dims": self.MACHINE_DIMS,
         }
 
-        results["revision"] = self._get_revision(uid, instrument)
-        revision = results["revision"]
+        results["revision"] = self._get_revision(uid, instrument, revision)
 
         location, location_path = self._get_signal(
-            uid, instrument, ".geometry:location", revision
+            uid, instrument, ".geometry:location", results["revision"]
         )
         direction, direction_path = self._get_signal(
-            uid, instrument, ".geometry:direction", revision
+            uid, instrument, ".geometry:direction", results["revision"]
         )
         if len(np.shape(location)) == 1:
             location = np.array([location])
             direction = np.array([direction])
 
-        results["t"], _ = self._get_signal(uid, instrument, ":time", revision)
-        wavelength, _ = self._get_signal(uid, instrument, ":wavelen", revision)
+        results["t"], _ = self._get_signal(
+            uid, instrument, ":time", results["revision"]
+        )
+        wavelength, _ = self._get_signal(
+            uid, instrument, ":wavelength", results["revision"]
+        )
+        # TODO: change once wavelength in MDS+ has been fixed to nanometers!
```

Comment:
Why has this been readded? New XRCS runs don't need this.

Reply:
Error: will check it out and fix. Send it back you you by EOP today